lib/galaxy/datatypes/data.py

```diff
@@ -355,11 +359,8 @@ def display_data(self, trans, data, preview=False, filename=None, to_ext=None, *
                 return self._archive_composite_dataset( trans, data, **kwd )
             else:
                 trans.response.headers['Content-Length'] = int( os.stat( data.file_name ).st_size )
-                if not to_ext:
-                    to_ext = data.extension
-                fname = ''.join(c in FILENAME_VALID_CHARS and c or '_' for c in data.name)[0:150]
-                trans.response.set_content_type( "application/octet-stream" )  # force octet-stream so Safari doesn't append mime extensions to filename
```

Comment:
https://github.com/galaxyproject/galaxy/pull/4004 - sorry about that and thanks for the catch.